File: opcua-webservice-spring/core/src/main/java/org/opcfoundation/webserver/addressspace/nodes/builtin/UaReferenceTypes.java
package org.opcfoundation.webserver.addressspace.nodes.builtin;

import org.eclipse.milo.opcua.stack.core.NodeIds;
import org.eclipse.milo.opcua.stack.core.types.builtin.LocalizedText;
import org.eclipse.milo.opcua.stack.core.types.builtin.NodeId;
import org.opcfoundation.webserver.addressspace.nodes.UaReferenceType;

public class UaReferenceTypes {
    public final static UaReferenceType References = new UaReferenceType(NodeIds.References,"References", new LocalizedText("References"), true,LocalizedText.NULL_VALUE, true);
    public final static UaReferenceType HierarchicalReferences = new UaReferenceType(NodeIds.HierarchicalReferences,"HierarchicalReferences", new LocalizedText("HierarchicalReferences"), true,new LocalizedText("InverseHierarchicalReferences"), false);
    public final static UaReferenceType NonHierarchicalReferences = new UaReferenceType(NodeIds.NonHierarchicalReferences,"NonHierarchicalReferences", new LocalizedText("NonHierarchicalReferences"), true, LocalizedText.NULL_VALUE, true);

    public final static UaReferenceType Organizes = new UaReferenceType(NodeIds.Organizes,"Organizes", new LocalizedText("Organizes"), false,new LocalizedText("OrganizedBy"), false);
    public final static UaReferenceType HasChild = new UaReferenceType(NodeIds.HasChild,"HasChild", new LocalizedText("HasChild"), true,new LocalizedText("ChildOf"), false);
    public final static UaReferenceType HasSubType = new UaReferenceType(NodeIds.HasSubtype,"HasSubtype", new LocalizedText("HasSubtype"), false,new LocalizedText("SubtypeOf"), false);
    public final static UaReferenceType Aggregates = new UaReferenceType(NodeIds.Aggregates,"Aggregates", new LocalizedText("Aggregates"), true,new LocalizedText("AggregatedBy"), false);
    public final static UaReferenceType HasComponent = new UaReferenceType(NodeIds.HasComponent,"HasComponent", new LocalizedText("HasComponent"), false,new LocalizedText("ComponentOf"), false);
    public final static UaReferenceType HasProperty = new UaReferenceType(NodeIds.HasProperty,"HasProperty", new LocalizedText("HasProperty"), false,new LocalizedText("PropertyOf"), false);

    public final static UaReferenceType HasTypeDefinition = new UaReferenceType(NodeIds.HasTypeDefinition,"HasTypeDefinition", new LocalizedText("HasTypeDefinition"), false,new LocalizedText("TypeDefinitionOf"), false);
    public final static UaReferenceType HasModellingRule = new UaReferenceType(NodeIds.HasModellingRule,"HasModellingRule", new LocalizedText("HasModellingRule"), false,new LocalizedText("ModellingRuleOf"), false);

    public final static UaReferenceType HasLink = new UaReferenceType(new NodeId(1,1),"HasLink", new LocalizedText("HasLink"), false,new LocalizedText("HasLink"), true);
}

File: opcua-webservice-node/src/server/addressspace/nodes/builtin/UaReferenceTypes.ts
import { ReferenceTypeIds, UaLocalizedText, UaNodeId } from "opcua-webapi-ts";
import { UaReferenceType } from "../UaReferenceType";

export class UaReferenceTypes {
    static References = new UaReferenceType(UaNodeId.from(ReferenceTypeIds.References),"References",new UaLocalizedText("References"),true,UaLocalizedText.nullText,true);
    static HierarchicalReferences = new UaReferenceType(UaNodeId.from(ReferenceTypeIds.HierarchicalReferences),"HierarchicalReferences",new UaLocalizedText("HierarchicalReferences"),true,new UaLocalizedText("InverseHierarchicalReferences"),false);
    static NonHierarchicalReferences = new UaReferenceType(UaNodeId.from(ReferenceTypeIds.NonHierarchicalReferences),"NonHierarchicalReferences",new UaLocalizedText("NonHierarchicalReferences"),true,UaLocalizedText.nullText,true);

    static Organizes = new UaReferenceType(UaNodeId.from(ReferenceTypeIds.Organizes),"Organizes",new UaLocalizedText("Organizes"),false,new UaLocalizedText("OrganizedBy"),false);
    static HasChild = new UaReferenceType(UaNodeId.from(ReferenceTypeIds.HasChild),"HasChild",new UaLocalizedText("HasChild"),true,new UaLocalizedText("ChildOf"),false);
    static HasSubtype = new UaReferenceType(UaNodeId.from(ReferenceTypeIds.HasSubtype),"HasSubtype",new UaLocalizedText("HasSubtype"),false,new UaLocalizedText("SubtypeOf"),false);
    static Aggregates = new UaReferenceType(UaNodeId.from(ReferenceTypeIds.Aggregates),"Aggregates",new UaLocalizedText("Aggregates"),true,new UaLocalizedText("AggregatedBy"),false);
    static HasComponent = new UaReferenceType(UaNodeId.from(ReferenceTypeIds.HasComponent),"HasComponent",new UaLocalizedText("HasComponent"),false,new UaLocalizedText("ComponentOf"),false);
    static HasProperty = new UaReferenceType(UaNodeId.from(ReferenceTypeIds.HasProperty),"HasProperty",new UaLocalizedText("HasProperty"),false,new UaLocalizedText("PropertyOf"),false);

    static HasTypeDefinition = new UaReferenceType(UaNodeId.from(ReferenceTypeIds.HasTypeDefinition),"HasTypeDefinition",new UaLocalizedText("HasTypeDefinition"),false,new UaLocalizedText("TypeDefinitionOf"),false);
    static HasModellingRule = new UaReferenceType(UaNodeId.from(ReferenceTypeIds.HasModellingRule),"HasModellingRule",new UaLocalizedText("HasModellingRule"),false,new UaLocalizedText("ModellingRuleOf"),false);
    static GeneratesEvent = new UaReferenceType(UaNodeId.from(ReferenceTypeIds.GeneratesEvent),"GeneratesEvent",new UaLocalizedText("GeneratesEvent"),false,new UaLocalizedText("GeneratedBy"),false);
    static HasLink = new UaReferenceType(new UaNodeId(1,1),"HasLink",new UaLocalizedText("HasLink"),false,new UaLocalizedText("HasLink"),true);
}
